mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/model/spi/GenericContextAwarePojoPropertyMode.java

```diff
@@ -43,7 +43,7 @@ public String getName() {
 	}
 
 	@Override
-	public PropertyHandle getHandle() {
+	public PropertyHandle<? super T> getHandle() {
 		return rawPropertyModel.getHandle();
```

Comment:
This should really return `PropertyHandle<T>`.

The thing is, the "raw property model" represents the property with its raw type, and this class represents the property with generics.

Let's say the property has type `List<U>` and is declared in a class `MyEntity<U>`. The raw property model will represent a property
of type `List`. A generic context aware propery model in the context of `MyEntity<Integer>` will represent a property of type `List<Integer>`.
So you will have a `GenericContextAwarePojoPropertyModel<List<Integer>>` which contains a rawPropertyModel of type `PojoPropertyModel<? super List<Integer>>` which is really a `PojoPropertyModel<List>`.
And we know that, in the context of `MyEntity<List<Integer>>`, the values contained in the property *can* be casted from `List` to `List<Integer>`. We have this knowledge right here, in this very class.

So this is the right place to perform the cast:

```
	@Override
	@SuppressWarnings("unchecked") // We know that, in the current generic context, this cast is legal
	public PropertyHandle<T> getHandle() {
		return (PropertyHandle<T>) rawPropertyModel.getHandle();
	}
```